src/modules/main/sections/TableDescription.vue

```diff
@@ -0,0 +1,149 @@
+<template>
+	<div class="element-description">
+		<div class="mode-switch">
+			<div class="buttons-wrap">
+				<NcButton v-if="mode !== 'edit' && canManageElement(activeElement)" @click="() => mode='edit'">
+					<template #icon>
+						<IconPencil :size="15" />
+					</template>
+					<template #default>
+						{{ t('tables','Edit') }}
+					</template>
+				</NcButton>
+				<NcButton v-if="mode !== 'view'" :size="15" @click="() => mode='view'">
+					<template #icon>
+						<IconCheck :size="15" />
+					</template>
+					<template #default>
+						{{ t('tables','Done') }}
+					</template>
+				</NcButton>
+			</div>
+		</div>
+		<div v-show="mode !== 'hidden'" class="description__editor">
+			<div id="description-editor" ref="textEditor" />
+		</div>
+	</div>
+</template>
+
+<script>
+
+import { NcButton } from '@nextcloud/vue'
+import permissionsMixin from '../../../shared/components/ncTable/mixins/permissionsMixin.js'
+import IconPencil from 'vue-material-design-icons/Pencil.vue'
+import IconCheck from 'vue-material-design-icons/Check.vue'
+
+export default {
+	name: 'TableDescription',
+
+	components: {
+		NcButton,
+		IconPencil,
+		IconCheck,
+	},
+	mixins: [permissionsMixin],
+	props: {
+		activeElement: {
+			type: Object,
+			default: null,
+		},
+	},
+
+	data() {
+		return {
+			mode: 'view',
+			description: '',
+		}
+	},
+	watch: {
+		mode() {
+			this.editor.setReadOnly(this.mode === 'view')
+		},
+	},
+
+	mounted() {
+		this.setupEditor()
+	},
+	async beforeDestroy() {
+		await this.destroyEditor()
+	},
+	methods: {
+		async setupEditor() {
+			if (this?.editor) await this.destroyEditor()
+			this.descriptionLastEdited = 0
+			this.description = this.activeElement.description
+			if (this.$refs.textEditor === undefined) {
+				return
+			}
+			this.editor = await window.OCA.Text.createEditor({
+				el: this.$refs.textEditor,
+				content: this.activeElement.description,
+				readOnly: !this.canManageElement(this.activeElement),
+				onUpdate: ({ markdown }) => {
+					if (this.description === markdown) {
+						this.descriptionLastEdit = 0
+						return
+					}
+					this.description = markdown
+					this.updateDescription()
+				},
+			})
+			this.editor.setReadOnly(true)
+		},
+		async saveDescription() {
+			if (this.descriptionLastEdited !== 0 || this.description === this.activeElement.description || !this.canManageElement(this.activeElement)) return
```

Comment:
Generally preferred code style around the Nextcloud code base and in this case also more readable I think ;)
```suggestion
			if (this.descriptionLastEdited !== 0 || this.description === this.activeElement.description || !this.canManageElement(this.activeElement)) {
				return
			}
```